tests/cleanup_and_retention_test.go

```diff
@@ -1,191 +1,166 @@
-			require.NoError(t, err)
+			values = append(values, row{id[:], startMilli + int64(j)})
 
-			total := scenario.rowsToDelete + scenario.rowsToSpare
-			type row struct {
-				Id   int64
-				Time int64
+			if len(values) == MaxPlaceholders || j == (rowsToDelete+rowsToSpare-1) {
+				_, err := db.NamedExec(fmt.Sprintf(`INSERT INTO %s (%s, %s) VALUES (:id, :time)`, stmt.Table, stmt.PK, stmt.Column), values)
+				require.NoError(t, err)
+				values = values[:0]
 			}
-			values := make([]row, 0, MaxPlaceholders)
-			start := time.Now().AddDate(0, 0, -retentionDays).Add(-1 * time.Millisecond * time.Duration(scenario.rowsToDelete))
-			startMilli := start.UnixMilli()
-			for j := int64(0); j < total; j++ {
-				if j == scenario.rowsToDelete {
-					startMilli += int64(2 * time.Minute)
-				}
-
-				values = append(values, row{j, startMilli + j})
-
-				if len(values) == MaxPlaceholders || j == total-1 {
-					_, err := db.NamedExec(fmt.Sprintf(`INSERT INTO %s (%s, %s) VALUES (:id, :time)`, stmt.Table, stmt.PK, stmt.Column), values)
-					require.NoError(t, err)
-					values = values[:0]
-				}
-			}
-
-			wg.Add(1)
 		}
+	}
 
-		ctx, cancelCtx := context.WithCancel(context.Background())
-		type result struct {
-			table   string
-			cleanup icingadb.CleanupResult
-		}
-		results := make(chan result)
-		go func() {
-			defer close(results)
-
-			ret := history.NewRetention(db, uint64(retentionDays), time.Hour, uint64(scenario.limit), history.RetentionOptions{}, logging.NewLogger(it.Logger(t).Sugar(), time.Second))
-			err := ret.StartWithCallback(ctx, func(t string, rs icingadb.CleanupResult) {
-				select {
-				case results <- result{t, rs}:
-				case <-ctx.Done():
-				}
-			})
-			if !errors.Is(err, context.Canceled) {
-				t.Error(err)
-			}
-		}()
-
-		cleanups := make(map[string]icingadb.CleanupResult, len(history.RetentionStatements))
-		timeout := time.After(time.Minute)
-		for done := false; !done; {
-			select {
-			case rs := <-results:
-				if _, ok := cleanups[rs.table]; ok {
-					t.Error("cleanup result for table", rs.table, "already exists")
-					done = true
-					break
-				}
-
-				cleanups[rs.table] = rs.cleanup
-
-				if len(cleanups) == len(history.RetentionStatements) {
-					done = true
-					break
-				}
-			case <-timeout:
-				t.Error("timeout exceeded")
-				done = true
-				break
+	r := it.RedisServerT(t)
+	i := it.Icinga2NodeT(t, "master")
+	i.EnableIcingaDb(r)
+	i.Reload()
+	waitForDumpDoneSignal(t, r, 20*time.Second, 100*time.Millisecond)
+	config, err := yaml.Marshal(struct {
+		Retention retention `yaml:"history-retention"`
+	}{reten})
+	assert.NoError(t, err)
+	it.IcingaDbInstanceT(t, r, rdb, services.WithIcingaDbConfig(string(config)))
+
+	eventually.Assert(t, func(t require.TestingT) {
+		for category, stmt := range retentionStatements {
+			var count int
+			err := db.QueryRow(fmt.Sprintf(`SELECT COUNT(*) FROM %s`, stmt.Table)).Scan(&count)
```

Comment:
Some check that the oldest rows were indeed the ones deleted would be nice. So basically doing that query twice, once with `WHERE time_column < threshold` and once with the inverse.

Reply:
Done.